bugbug/model.py

```diff
@@ -567,8 +569,24 @@ def train(self, importance_cutoff=0.15, limit=None):
 
             self.clf.fit(X_train, self.le.transform(y_train))
 
-        with open(self.__class__.__name__.lower(), "wb") as f:
-            pickle.dump(self, f, protocol=pickle.HIGHEST_PROTOCOL)
+        model_name = self.__class__.__name__.lower()
+
+        if issubclass(type(self.clf), XGBModel):
+            # Since the workflow expects a single file, we create a directory
+            # with the expected file name to store multiple model-related files.
+            os.makedirs(model_name, exist_ok=True)
+
+            self.clf.save_model(os.path.join(model_name, "xgboost.ubj"))
+
+            # Since we save the classifier separately, we need to clear the clf
+            # attribute to prevent it from being pickled with the model object.
+            self.clf = None
```

Comment:
Using `Booster.load_model()` could cause problems when using it with Scikit-learn API.[^1] Thus, instead of replacing `self.clf` with `None`, we could replace it with any empty class. Then, when loading the mode, we could use the `self.clf.load_model()` method for that.

[^1]: https://mljar.com/blog/xgboost-save-load-python/#save-the-xgboost-booster-object